src/manifest-converter.ts

```diff
@@ -92,55 +93,155 @@ export function writeJson(json: any, ...pathPieces: string[]) {
 }
 
 /**
- * Given a bower.json manifest, generate a package.json manifest for npm.
+ * Generate the package.json for a modulized package from its bower.json,
+ * optionally merging with an existing package.json.
  *
- * Function takes an optional `useLocal` argument containing a map of any
- * npm dependencies (name -> local file path) that should be referenced via
- * local file path and not public package name in the package.json. This is
- * useful for testing against other, converted repos.
+ * @param bowerJson The package's existing parsed bower.json.
+ * @param name NPM package name. Always wins over existingPackageJson.
+ * @param version NPM package version. Always wins over existingPackageJson.
+ * @param useLocal Optional map of any NPM dependencies (name -> local file
+ * path) that should be referenced via local file path and not public package
+ * name in the package.json. This is useful for testing against other, converted
+ * repos.
+ * @param existingPackageJson Optional pre-existing parsed package.json. If
+ * provided, values from this package.json will win over ones derived from
+ * bower.json, with these exceptions:
+ *   - name, version, flat, and private are always overridden.
+ *   - dependencies, devDependencies, and resolutions are merged, with newly
+ *     generated versions for the same package winning.
  */
 export function generatePackageJson(
-    bowerJson: any,
-    npmName: string,
-    npmVersion: string,
-    useLocal?: Map<string, string>) {
-  const packageJson = {
-    name: npmName,
-    flat: true,
-    version: npmVersion,
+    bowerJson: Partial<BowerConfig>,
+    name: string,
+    version: string,
+    useLocal?: Map<string, string>,
+    existingPackageJson?: Partial<YarnConfig>): YarnConfig {
+  const packageJson: YarnConfig = {
     description: bowerJson.description,
-    author: bowerJson.author,
-    contributors: bowerJson.contributors || bowerJson.authors,
     keywords: bowerJson.keywords,
-    main: (typeof bowerJson.main === 'string') ? bowerJson.main : undefined,
     repository: bowerJson.repository,
-    license: bowerJson.license,
     homepage: bowerJson.homepage,
-    dependencies: <any>{},
-    devDependencies: <any>{},
-    // TODO (fks): Remove these resolutions needed by wct-browser-legacy
-    // https://github.com/Polymer/polymer-modulizer/issues/251
-    resolutions: {
-      'inherits': '2.0.3',
-      'samsam': '1.1.3',
-      'supports-color': '3.1.2',
-      'type-detect': '1.0.0',
-    }
+
+    ...existingPackageJson,
+
+    name,
```

Comment:
maybe we should warn if the name doesn't match?

Reply:
Done.